DXMainClient/DXGUI/Multiplayer/GameLobby/MultiplayerGameLobby.cs

```diff
@@ -197,6 +205,23 @@ public override void Initialize()
             // To move the lblMapAuthor label into its correct position
             // if it was moved in the theme description INI file
             LoadDefaultMap();
+            GetGameModeAliases();
+        }
+
+        private void GetGameModeAliases()
```

Comment:
I moved LoadCustomMap to MapLoader. I also added the messages to print into chatbox when using the chat command to load custom maps and not just when map sharing. Not sure if this was disabled on purpose. I could not figure out a reason for it to be, since the messages do not get displayed to everyone anyway. If it is a problem, it can easily be changed back .

Now there might be an another problem, however. As it stands, and did so before any changes I did, loading a custom map through chat command did absolutely nothing besides adding the map to gamemode's list of maps, whilst map sharing, if successful, changes the map, but does not refresh the map list so the change is not reflected in there. I am not sure what the best approach here would be, I figure safest route would be to refresh the map list so the newly added custom map appears in there but stick to the currently selected map and game mode, but I suppose it would be better if the client was able to reliably swap to the new map and gamemode, if it was properly added to the list and reflect the change in UI as well.

Reply:
Yeah, the command added the map to the game mode's list of maps. The map would then be listed upon changing the game mode from the drop-down. It'd of course be better to also refresh the map list. I'm not sure if it should also swap to the newly-added map right away.

Map sharing does not affect the game host's map list at all and non-host players don't have the map list in their UI, so that's why it doesn't refresh the map list.